sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -242,6 +246,58 @@ private static synchronized void init(
     notifyOptionsObservers(options);
 
     finalizePreviousSession(options, HubAdapter.getInstance());
+
+    handleStartupProfilingConfig(options, options.getExecutorService());
+  }
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private static void handleStartupProfilingConfig(
+      final @NotNull SentryOptions options,
+      final @NotNull ISentryExecutorService sentryExecutorService) {
+    sentryExecutorService.submit(
+        () -> {
+          final String cacheDirPath = options.getCacheDirPathWithoutDsn();
+          if (cacheDirPath != null) {
+            final @NotNull File startupProfilingConfigFile =
+                new File(cacheDirPath, STARTUP_PROFILING_CONFIG_FILE_NAME);
+            // We always delete the config file for startup profiling
+            FileUtils.deleteRecursively(startupProfilingConfigFile);
+            if (!options.isEnableStartupProfiling()) {
+              return;
+            }
+            if (!options.isTracingEnabled()) {
+              options
+                  .getLogger()
+                  .log(
+                      SentryLevel.INFO,
+                      "Tracing is disabled and startup profiling will not start.");
+              return;
+            }
+            try {
+              if (startupProfilingConfigFile.createNewFile()) {
+                final @NotNull TracesSamplingDecision startupSamplingDecision =
+                    sampleStartupProfiling(options);
+                final @NotNull SentryStartupProfilingOptions startupProfilingOptions =
+                    new SentryStartupProfilingOptions(options, startupSamplingDecision);
+                try (Writer fileWriter = new SentryFileWriter(startupProfilingConfigFile)) {
+                  options.getSerializer().serialize(startupProfilingOptions, fileWriter);
+                }
+              }
+            } catch (IOException e) {
+              options
+                  .getLogger()
+                  .log(SentryLevel.ERROR, "Unable to create startup profiling config file. ", e);
+            }
+          }
+        });
+  }
+
+  private static @NotNull TracesSamplingDecision sampleStartupProfiling(
+      final @NotNull SentryOptions options) {
+    TransactionContext startupTransactionContext = new TransactionContext("ui.load", "");
```

Comment:
hm, should `ui.load` be an operation name, and not a transaction name? In that case it should be the 2nd parameter then. Also I don't know how the frontend UI will look like for this, but don't we also need a meaningful transaction name?

Reply:
ok, I've just checked the 3rd PR, I guess it doesn't really matter what we use here, because the profile is attached to the first activity tx anyway, right?

Reply:
ok, I see you've changed it there, resolving this

Reply:
yep. Also, the name and operation will be "app.launch" and "profile" (in third PR), which doesn't really matter, as the user should just check the "isForNextStartup" flag.
But with those values we aligned Android and Cocoa